linalg/blockvector.cpp

```diff
@@ -90,7 +90,11 @@ void BlockVector::Update(const Array<int> &bOffsets, bool force)
             for (int i = 0; true; i++)
             {
                if (blockOffsets[i] != bOffsets[i]) { break; }
-               if (i == numBlocks) { return; }
+               if (i == numBlocks)
+               {
+                  blockOffsets = bOffsets.GetData();
```

Comment:
Wouldn't that leak memory?

Reply:
The pointer `blockOffsets` is not owned by the class, so this is OK.